redisraft.h

```diff
@@ -215,6 +215,18 @@ extern raft_log_impl_t RaftLogImpl;
 #define REDIS_RAFT_DEFAULT_LOG_MAX_CACHE_SIZE       8*1000*1000
 #define REDIS_RAFT_DEFAULT_LOG_MAX_FILE_SIZE        64*1000*1000
 
+#define REDIS_RAFT_HASH_SLOTS                       16384
+#define REDIS_RAFT_HASH_MIN_SLOT                    0
+#define REDIS_RAFT_HASH_MAX_SLOT                    16383
+
+#define REDIS_RAFT_VALID_HASH_SLOT(h)   \
+    ((h) >= REDIS_RAFT_HASH_MIN_SLOT && (h) <= REDIS_RAFT_HASH_MAX_SLOT)
+
+#define REDIS_RAFT_VALID_HASH_SLOT_RANGE(a, b) \
+    (REDIS_RAFT_VALID_HASH_SLOT(a) && \
+     REDIS_RAFT_VALID_HASH_SLOT(b) && \
+     (a) <= (b))
```

Comment:
Do we really need to do this with macros in 2020? Why not use functions instead? The compiler should inline them anyway.

Reply:
All the reasons against using macros, except for habit. Will do.